lib/IRGen/GenDistributed.cpp

```diff
@@ -647,11 +642,8 @@ void DistributedAccessor::emit() {
   // Metadata that represents passed in the invocation decoder.
   auto *decoderType = params.claimNext();
 
-  // If the distributed thunk is declared in a protocol that conforms
-  // to `DistributedActor` protocol, there is an extract parameter that
-  // represents a type of protocol witness.
-  if (isa<ProtocolDecl>(actor))
-    (void)params.claimNext();
+  // Metadata of the distributed actor the accessor is for.
+  (void)params.claimNext();
```

Comment:
I think we might want to change ordering of this because decoder is now first generic parameter and actor follows it, if I remember correctly witnesses are passed in order, that's probably what is causing all these nullptr issues.

Reply:
Or we just make it so the actor is the first parameter, that works too.

Reply:
Okey I'll give either of those a try